File: StockChess/StockChessCS/Converters/FileColumnConverter.cs
using System;
using System.Linq;
using System.Globalization;
using System.Windows.Data;

namespace StockChessCS.Converters
{
    public class FileColumnConverter : IValueConverter
    {
        public object Convert(object value, Type targetType, object parameter, CultureInfo culture)
        {
            var file = (char) value;
            var files = ("abcdefgh").ToArray();
            var column = files.ToList().IndexOf(file);
            return column;
        }

        public object ConvertBack(object value, Type targetType, object parameter, CultureInfo culture)
        {
            return Binding.DoNothing;
        }
    }
}

File: StockChess/StockChess/Converters/FileColumnConverter.vb
Imports System.Globalization

Public Class FileColumnConverter
    Implements IValueConverter

    Public Function Convert(value As Object, targetType As Type, parameter As Object, culture As CultureInfo) As Object Implements IValueConverter.Convert
        Dim file = CChar(value)
        Dim files = ("abcdefgh").ToArray()
        Dim column = files.ToList.IndexOf(file)
        Return column
    End Function

    Public Function ConvertBack(value As Object, targetType As Type, parameter As Object, culture As CultureInfo) As Object Implements IValueConverter.ConvertBack
        Return Binding.DoNothing
    End Function
End Class
